### PR TITLE
Approve member create company

### PR DESCRIPTION
**Pull request by weedySeaDragon**
PT Story:  **When a member is approved, create the company, change their status, enter a membership number**
https://www.pivotaltracker.com/story/show/135472437

As an admin
so that a new member gets notified that they have been approved and can then fill out their info,
when I change their status to approved,
create their Company if it doesn't already exist,
and associate them with the company,
and then l'll manually enter a new membership number for them.

This all takes place in the `MembershipApplicationController`. 
Right now we are **_not_** using Javascript on the UI to see if the admin has change the drop-down list for the status.  So we make them go through a 2 step process:  change the status to `Accepted` and save.  We check for that (in `update`). If they did, then we what is needed with the models, etc.  Then we redirect them to the edit page and have them enter a (new) `membership number` and then save again.  It's not smooth, but for MVP, that's fine.
Hence the `redirect ..... return` statement in `update`.

**Simple scenario is working! (status is changed, a new company is created)**
Also working: If a member is approved, and then the status is later changed to 'Rejected,' the member number no longer appears.  (If any company was created, that remains; it's not deleted.)

### I cannot get 1 scenario to pass.
  However, when I do it for real, it works.  Something must be wrong with the feature (the scenario, the steps, the factories...?)
The offending scenario is **Scenario: Admin approves, member is added to existing company** in `approve-member.feature`

---
Because this is the core functionality for the admin, this touches on all models and exercises the system.  I found some issues.

Changes proposed in this pull request:
1. We currently expect the status to == "Accepted".  It was still 'approved' in some features/scenarios.
2. Renamed `@business_categories` to `@all_business_categories` in appropriate places to reduce confusion 
3. New ApplicationHelper method `assocation_empty?` because ActiveRecord assocations are really proxies.  Data isn't always loaded. Was important when checking to see if a collection was empty.
4.  Added `member_number` to MembershipApplicationPolicy so it is accepted as a (strong) parameter.  Renamed methods in that class to reflect that addition
5. Logic as explained above to change the member status to accepted.

Ready for review:
@

### CODE REVIEW

File: app/models/company.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class Company < ApplicationRecord
 
   include HasSwedishOrganization
 
-  validates_presence_of :name, :company_number
+  validates_presence_of :company_number
   validates_presence_of :company_number
```
Comment (thesuss) [inline app/models/company.rb]: is it needed twice?
Or maybe: Why is name removed?

Comment (weedySeaDragon) [reply app/models/company.rb]: A user is not required to provide a company name when they fill out their application.  So when an admin approves it and we have to instantiate a company, we don't have a name.  
IMHO, the right thing to do is to require the user to provide us with a company name when they fill out the application.  (but after MVP)

File: features/admin_creates_company.feature [hunk 2]
```diff
@@ -60,7 +60,6 @@ Feature: As an admin
     Scenarios:
       | name        | org_number | phone      | street         | post_code | city   | region    | email                | website                   | error                                                          |
       | Happy Mutts | 00         | 0706898525 | Ålstensgatan 4 | 123 45    | Bromma | Stockholm | kicki@gladajyckar.se | http://www.gladajyckar.se | "Company number is the wrong length (should be 10 characters)" |
-      |             | 5562252998 |            | Ålstensgatan 4 | 123 45    | Bromma | Stockholm | kicki@gladajyckar.se | http://www.gladajyckar.se | "Name can't be blank"                                          |
```
Comment (weedySeaDragon) [inline features/admin_creates_company.feature:63]: Because now the name *can* be blank.

File: app/models/company.rb (hunk 1 above)
Comment (thesuss) [reply app/models/company.rb]: Understand :) thanks

Comment (thesuss) [inline app/models/company.rb]: Check slack ;)

Comment (thesuss) [inline app/models/company.rb]: @weedySeaDragon check slack :)

**Review by thesuss — APPROVED** [review]

**Pull request merged**